pjsip/src/pjsip/sip_transport_tcp.c

```diff
@@ -401,8 +401,7 @@ PJ_DEF(pj_status_t) pjsip_tcp_transport_start3(
     listener->initial_timeout = cfg->initial_timeout;
     pj_memcpy(&listener->qos_params, &cfg->qos_params,
               sizeof(cfg->qos_params));
-    pj_memcpy(&listener->sockopt_params, &cfg->sockopt_params,
-              sizeof(cfg->sockopt_params));
+    pj_sockopt_params_clone(&listener->sockopt_params, &cfg->sockopt_params);
```

Comment:
Why? I think the deep clone is needed here since the config is supplied by app. The one that's not needed is in `sip_transport_tls` when creating a transport from listener, since the listener should be available during the transport's lifetime.